manifests/crds/application-crd.yaml

```diff
@@ -1851,7 +3106,11 @@ spec:
                 type: array
               sourceType:
                 description: SourceType specifies the type of this application
-                type: string
+                items:
+                  description: ApplicationSourceType specifies the type of the application's
+                    source
+                  type: string
+                type: array
```

Comment:
This caused the controller to crash when handling Applications that already had this field populated. 

```
time="2022-09-16T20:26:27Z" level=info msg="ArgoCD Application Controller is starting" built="2022-09-16T19:24:57Z" commit=ce5a3b0b16cffe3af3dd3f79ea14fda4d07aeef5 namespace=argocd version=v2.5.0+ce5a3b0.dirty
time="2022-09-16T20:26:27Z" level=info msg="Processing all cluster shards"
time="2022-09-16T20:26:27Z" level=info msg="appResyncPeriod=3m0s, appHardResyncPeriod=0s"
time="2022-09-16T20:26:27Z" level=info msg="Starting configmap/secret informers"
W0916 20:26:27.344820       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
E0916 20:26:27.344881       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Application: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
time="2022-09-16T20:26:27Z" level=info msg="Configmap/secret informer synced"
time="2022-09-16T20:26:27Z" level=info msg="Ignore status for CustomResourceDefinitions"
time="2022-09-16T20:26:27Z" level=info msg="Ignore '/spec/preserveUnknownFields' for CustomResourceDefinitions"
W0916 20:26:28.773041       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
E0916 20:26:28.773114       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Application: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
W0916 20:26:30.703370       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
E0916 20:26:30.703416       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Application: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
W0916 20:26:37.008324       1 reflector.go:324] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
E0916 20:26:37.008374       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Application: failed to list *v1alpha1.Application: json: cannot unmarshal string into Go struct field ApplicationStatus.items.status.sourceType of type []v1alpha1.ApplicationSourceType
```